src/ir/LocalGraph.cpp

```diff
@@ -50,7 +50,7 @@ struct Action {
 // information about a basic block
 struct Info {
   std::vector<Action> actions; // actions occurring in this block
-  std::vector<SetLocal*> lastSets; // for each index, the last set_local for it
+  std::unordered_map<Index, SetLocal*> lastSets; // for each index, the last set_local for it
```

Comment:
why is this an unordered map, while in `OptimizedBlock` we use a very different data structure? if the perf aspects justify the difference, please explain in comments why.

Reply:
There it's still an unordered_map from my previous [commit](https://github.com/WebAssembly/binaryen/pull/1610/commits/f8e89b0934c97d65aeecdc353ca03a4a02af7b50). Indeed before It was a vector containing the lastSets per cells (each cell were an Index => Bad idea when there are a lot of locals).
In this commit, changing this container speed up the process.

When I changed the algorithm to introduce the new struct, I noticed that this container wasn't so big (worst case one hundred elements) and profiler told me that searching take some times. So I test using a vector which are appropriate there.

I keep the unordered_map because it's easy to gather last item for a key using a map :)

When you speak about comments, there or in the source code ?